File: python/01_slack.py
import json
import os
import urllib.request

def lambda_handler(event, context):
    notify('Hello World Python')
    return {
        'statusCode': 200,
        'body': json.dumps('Hello from Lambda!')
    }

def notify(text):
    data = json.dumps({'text': text}).encode('utf-8')
    headers = {'Content-Type': 'application/json'}
    req = urllib.request.Request(
        os.environ['WEBHOOK_URL'],
        data=data,
        method='POST',
        headers=headers)
    with urllib.request.urlopen(req) as res:
        print(res.read().decode('utf-8'))

File: node/01_slack.js
const https = require('https')

exports.handler = async (event) => {
    await notify('Hello World Node')
    const response = {
        statusCode: 200,
        body: JSON.stringify('Hello from Lambda!'),
    }
    return response
}

const notify = text => {
    return new Promise((resolve, reject) => {
        const data = JSON.stringify({ text: text })
        const options = {
            method: 'POST',
            headers: {
                'Content-Type': 'application/json'
            }
        }
        const req = https.request(process.env['WEBHOOK_URL'], options, res => {
            console.log(res.statusCode)
            resolve()
        })
        req.write(data)
        req.end()
    })
}
